apps/financial-aid/web-veita/src/components/ModalTypes/AcceptModal.tsx

```diff
@@ -1,54 +1,337 @@
+  personalTaxCreditPercentage: number
+  tax: number
+  secondPersonalTaxCredit: number
+  showSecondPersonalTaxCredit: boolean
+  hasError: boolean
+  deductionFactor: Record<string, { description: string; amount: number }>
 }
 
 const AcceptModal = ({
   onCancel,
   onSaveApplication,
   isModalVisable,
+  homeCircumstances,
+  spouseNationalId,
 }: Props) => {
   const maximumInputLength = 6
-  const [amount, setAmount] = useState<number>(0)
-  const [hasError, setHasError] = useState(false)
+
+  const currentYear = format(new Date(), 'yyyy')
+
+  const { municipality } = useContext(AdminContext)
+
+  const aidAmount = useMemo(() => {
+    if (municipality && homeCircumstances) {
+      return aidCalculator(
+        homeCircumstances,
+        spouseNationalId
+          ? municipality.cohabitationAid
+          : municipality.individualAid,
+      )
+    }
+  }, [homeCircumstances, municipality])
+
+  if (!aidAmount) {
+    return (
+      <Text color="red400">
+        Útreikingur fyrir aðstoð misstókst, vinsamlegast reyndu aftur
+      </Text>
+    )
+  }
+
+  const [state, setState] = useState<calculationsState>({
+    amount: aidAmount,
+    income: 0,
+    personalTaxCreditPercentage: 0,
+    tax: calculateTaxOfAmount(aidAmount, currentYear),
+    secondPersonalTaxCredit: 0,
+    showSecondPersonalTaxCredit: false,
+    deductionFactor: {},
+    hasError: false,
+  })
+
+  const sumValues = (
+    obj: Record<string, { description: string; amount: number }>,
+  ) =>
+    Object.values(obj)
+      .map((item) => {
+        return item.amount
+      })
+      .reduce((a, b) => {
+        return a + b
+      }, 0)
+
+  const finalAmount = calculateAcceptedAidFinalAmount(
+    aidAmount - state.income - sumValues(state.deductionFactor),
+    currentYear,
+    state.personalTaxCreditPercentage,
+    state.secondPersonalTaxCredit,
+  )
 
   return (
     <InputModal
       headline="Umsóknin þín er samþykkt og áætlun er tilbúin"
       onCancel={onCancel}
       onSubmit={() => {
-        if (amount <= 0) {
-          setHasError(true)
+        if (finalAmount && finalAmount <= 0) {
+          setState({ ...state, hasError: true })
           return
         }
-        onSaveApplication(amount)
+        onSaveApplication(finalAmount)
       }}
       submitButtonText="Samþykkja"
       isModalVisable={isModalVisable}
-      hasError={hasError}
-      errorMessage="Þú þarft að setja inn upphæð"
+      hasError={state.hasError}
+      errorMessage="Þú þarft að fylla út alla reiti"
     >
-      <NumberInput
-        placeholder="Skrifaðu upphæð útborgunar"
-        id="amountInput"
-        name="amountInput"
-        value={amount.toString()}
-        onUpdate={(input) => {
-          setHasError(false)
-          setAmount(input)
-        }}
-        maximumInputLength={maximumInputLength}
-      />
+      <Box marginBottom={3}>
+        <NumberInput
+          label="Grunnupphæð"
+          placeholder="Skrifaðu upphæð útborgunar"
```

Comment:
Isn't it weird that the placeholder has 'skrifaðu' in it?

Here and in other places?